src/app/pages/dashboard/widgets/network/widget-network/widget-network.component.ts

```diff
@@ -0,0 +1,214 @@
+    shareReplay({ bufferSize: 1, refCount: true }),
+  );
+
+  protected interface = toSignal(this.interface$);
+
+  protected interfaceUsage = toSignal(this.resources.realtimeUpdates$.pipe(
+    skipWhile(() => Boolean(!this.interface()?.name)),
+    map((update) => update.fields.interfaces),
+    map((interfaces) => interfaces?.[this.interface().name]),
+  ));
+
+  protected reportingData = toSignal(this.interface$.pipe(
+    filter(Boolean),
+    switchMap((networkInterface) => this.resources.networkInterfaceUpdate(networkInterface.name)),
+    map((response) => {
+      const updatedResponse = response[0];
+      (updatedResponse.data as number[][]).forEach((row, index) => {
+        // remove first column and convert kilobits/s to bits/s
+        (updatedResponse.data as number[][])[index] = row.slice(1).map((value) => value * kb);
+      });
+      return updatedResponse;
+    }),
+  ));
+
+  protected isLinkStateUp = computed(() => {
+    if (this.interfaceUsage().link_state) {
+      return this.interfaceUsage().link_state === LinkState.Up;
+    }
+    return this.interface().state.link_state === LinkState.Up;
+  });
+
+  protected linkStateLabel = computed(() => {
+    if (this.interfaceUsage().link_state) {
+      return this.interfaceUsage().link_state.replace(/_/g, ' ');
+    }
+    return this.interface().state.link_state.replace(/_/g, ' ');
+  });
+
+  protected bitsIn = computed<number>(() => {
+    return this.interfaceUsage().received_bytes_rate * 8;
+  });
+
+  protected bitsOut = computed<number>(() => {
+    return this.interfaceUsage().sent_bytes_rate * 8;
+  });
+
+  protected chartData = computed<ChartData<'line'>>(() => {
+    const currentTheme = this.theme.currentTheme();
+    const response = this.reportingData();
+    const networkInterfaceName = this.interface().name;
+
+    const labels: number[] = (response.data as number[][]).map((_, index) => {
+      return (response.start + index) * 1000;
+    });
+
+    return {
+      datasets: [
+        {
+          label: this.translate.instant('Incoming [{networkInterfaceName}]', { networkInterfaceName }),
+          data: (response.data as number[][]).map((item, index) => ({ x: labels[index], y: item[0] })),
+          borderColor: currentTheme.blue,
+          backgroundColor: currentTheme.blue,
+          pointBackgroundColor: currentTheme.blue,
+          pointRadius: 0,
+          tension: 0.2,
+          fill: true,
+        },
+        {
+          label: this.translate.instant('Outgoing [{networkInterfaceName}]', { networkInterfaceName }),
+          data: (response.data as number[][]).map((item, index) => ({ x: labels[index], y: -item[1] })),
+          borderColor: currentTheme.orange,
+          backgroundColor: currentTheme.orange,
+          pointBackgroundColor: currentTheme.orange,
+          pointRadius: 0,
+          tension: 0.2,
+          fill: true,
+        },
+      ],
+    };
+  });
+
+  protected chartOptions: ChartOptions<'line'> = {
```

Comment:
We should break this large object down into smaller chunks stored in variables with declarative names so it's more readable what we're doing.

Reply:
It sounded like a good idea at first. When I did, it became less readable, and understanding what and where became more complex. So, I prefer to keep it as is. 

Here is what it might look like after splitting:
![image](https://github.com/truenas/webui/assets/351613/6fdeeb28-21e6-4f17-bfd3-a43c40795b9e)
![image](https://github.com/truenas/webui/assets/351613/150dbfe5-8986-482d-a325-b37bc0b3c3e3)